File: backend/src/main/java/com/cineplex/entity/Room.java
package com.cineplex.entity;

import com.cineplex.entity.enums.RoomStatus;
import com.cineplex.entity.enums.ScreenType;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "rooms", indexes = {
    @Index(name = "idx_room_status", columnList = "status")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(name = "total_rows", nullable = false)
    private Integer totalRows;

    @Column(name = "total_columns", nullable = false)
    private Integer totalColumns;

    @Enumerated(EnumType.STRING)
    @Column(name = "screen_type", nullable = false, length = 20)
    @Builder.Default
    private ScreenType screenType = ScreenType.STANDARD_2D;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private RoomStatus status = RoomStatus.ACTIVE;

    @OneToMany(mappedBy = "room", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Seat> seats = new ArrayList<>();
}

File: frontend/src/types/room.ts
export type ScreenType = 'STANDARD_2D' | 'THREE_D' | 'IMAX' | 'FOUR_DX';
export type RoomStatus = 'ACTIVE' | 'MAINTENANCE' | 'CLOSED';

export interface SeatType {
  id: number;
  name: string;
  surchargePrice: number;
  colorCode: string;
}

export interface Seat {
  id: number;
  roomId: number;
  seatType: SeatType;
  rowCode: string;
  colNumber: number;
  seatCode: string;
  isActive: boolean;
}

export interface Room {
  id: number;
  name: string;
  totalRows: number;
  totalColumns: number;
  totalSeats: number;
  screenType: ScreenType;
  status: RoomStatus;
  seats?: Seat[];
}
